Upgrade to Springboot 3.0.1

arvindkrishnakumar-okta opened the pull request:
Note: 

- Build will not be successful until https://github.com/okta/okta-oidc-tck/pull/179 gets merged. 
- JDK 8 & 11 builds will need to be removed from CCI (Springboot 3 [supports](https://spring.io/blog/2022/05/24/preparing-for-spring-boot-3-0) only JDK17+).

mraible commented:
What's the status of Spring Boot 3 support? I'm holding an updated blog post and screencast in anticipation of the next release!

arvindkrishnakumar-okta commented:
> What's the status of Spring Boot 3 support? I'm holding an updated blog post and screencast in anticipation of the next release!

I'm currently blocked on making Springboot 3 work with Wiremock. Trying to fix the test failures related to this. 
Ref: https://github.com/wiremock/wiremock/issues/1760

mraible commented:
@arvindkrishnakumar-okta https://github.com/wiremock/wiremock/issues/1760 issue is closed, so I'm not sure what the path forward is. Do the failing tests indicate that functionality is broken, or you're just unable to prove everything works with tests?

arvindkrishnakumar-okta commented:
> @arvindkrishnakumar-okta [wiremock/wiremock#1760](https://github.com/wiremock/wiremock/issues/1760) issue is closed, so I'm not sure what the path forward is. Do the failing tests indicate that functionality is broken, or you're just unable to prove everything works with tests?

Sorry, I got past that above issue. Now I am at a point where the test failures suggest that we're missing `org.springframework.security.oauth2.server.resource.web.BearerTokenAuthenticationFilter` in several IT flows.
Ref: https://app.circleci.com/pipelines/github/okta/okta-spring-boot/144/workflows/7f28a9bc-95df-4b55-81b0-e38038b0e576/jobs/325

mraible commented:
@arvindkrishnakumar-okta It looks like that class moved to `org.springframework.security.oauth2.server.resource.web.authentication.BearerTokenAuthenticationFilter`. 

The [Javadoc](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/oauth2/server/resource/web/BearerTokenAuthenticationFilter.html) for the old one points to the [above package](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/oauth2/server/resource/web/authentication/BearerTokenAuthenticationFilter.html).

arvindkrishnakumar-okta commented on pom.xml:
```diff
@@ -144,7 +144,7 @@
                 <groupId>com.okta.oidc.tck</groupId>
                 <artifactId>okta-oidc-tck</artifactId>
-                <version>0.5.9</version>
+                <version>0.5.10-SNAPSHOT</version>
```
TODO: Update after https://github.com/okta/okta-oidc-tck/pull/179 is merged.

```suggestion
                <version>0.5.10</version>
```

mraible commented:
FWIW, it seems like CI is still testing against Java 8 and Java 11. I'm not sure that'll work with Spring Boot 3 since Java 17 is the baseline. Also, it'd be awesome if the CI job could use `-ntp` and less logging to reduce the clutter.

arvindkrishnakumar-okta commented:
> FWIW, it seems like CI is still testing against Java 8 and Java 11. I'm not sure that'll work with Spring Boot 3 since Java 17 is the baseline. Also, it'd be awesome if the CI job could use `-ntp` and less logging to reduce the clutter.

Removed Java 8 & 11 from CCI. Added `ntp` to CCI.

mraible commented:
🚢 it!
